vscode/src/extension.ts

```diff
@@ -902,11 +902,42 @@ function doActivateWithJDK(specifiedJDK: string | null, context: ExtensionContex
         enableModules.push('org.netbeans.libs.nbjavacapi');
     }
 
+    let userHomeDir:string = '';
+    const isProjectFolderSearchLimited : boolean = !netbeansConfig.get('advanced.disable.projectSearchLimit', false);
+    if (isProjectFolderSearchLimited) {
+        try {
+            userHomeDir = os.homedir() as string;
+        } catch (err:any) {
+            handleLog(log, `Failed to obtain the user home directory due to: ${err}`);
+        }
+        const workspaces = workspace.workspaceFolders;
+        if (workspaces && workspaces[0]?.uri) {
+            if (userHomeDir) {
+                try {
+                    let projectFolderPath:string = path.normalize(workspaces[0].uri.fsPath);
+                    let homeDir:string = path.normalize(userHomeDir);
+                    if (os.type() === "Windows_NT" && homeDir.codePointAt(1) === 58) {
+                        homeDir = homeDir[0].toLowerCase() + homeDir.substring(1);  // Make drive letter lower-case in order to match fsPath.
+                    }
+                    if (!projectFolderPath.startsWith(path.normalize(homeDir))) {
+                        userHomeDir = path.normalize(workspaces[0].uri.path);   // Keep userHomeDir limited to the workspace path.
+                    }
+                } catch (err:any) {
+                    handleLog(log, `Failed to check whether the workspace is inside the userHomeDir: ${err}`);
+                }
+            }
+        } else if (!userHomeDir) {
+            userHomeDir = os.type() === "Windows_NT" ? '%USERPROFILE%' : '$HOME';   // The launcher script may perform the env variable substitution
```

Comment:
nitpick: Can we move `"Windows_NT"` to `constants.ts`, since it is used in #245 as well, so both can export from constants? (So one of the PRs may have to do this change, since I have made same comment on both the PRs).

Reply:
Makes sense. Thanks. I'll rebase once #245 is merged.